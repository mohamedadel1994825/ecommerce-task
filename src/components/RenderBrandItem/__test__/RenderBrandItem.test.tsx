import { Brand } from "@interfaces";
import { cleanup, fireEvent, render } from "@testing-library/react-native";
import React from "react";
import { RenderBrandItem } from "../index";

describe("RenderBrandItem", () => {
  afterEach(cleanup);

  const brand: Brand = { id: 1, name: "Test Brand" };
  const isSelected = true;
  const onPress = jest.fn();

  it("should match snapshot", () => {
    const { toJSON } = render(
      <RenderBrandItem item={brand} isSelected={isSelected} onPress={onPress} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render BrandItem component with correct props", () => {
    const { getByText } = render(
      <RenderBrandItem item={brand} isSelected={isSelected} onPress={onPress} />
    );
    const brandItem = getByText(brand.name);
    fireEvent.press(brandItem);
    expect(onPress).toHaveBeenCalled();
  });
});
