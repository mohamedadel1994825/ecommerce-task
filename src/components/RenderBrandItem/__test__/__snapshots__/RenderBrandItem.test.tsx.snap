// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderBrandItem should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#0040ff",
      "borderRadius": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="brand-item"
>
  <Text
    style={
      {
        "color": "#FFFFFF",
      }
    }
  >
    Test Brand
  </Text>
</View>
`;
